### ISSUE TITLE
Follow-up: Migration guides

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/docs#3819]
The changes from the new Migration Guides (detailed in #3766) were
merged to unblock other work, but they still need docs team review.

This issue is for tracking:

- Doing the docs team review(s)
- Making any necessary updates

**Comment by rmloveland** [comment]
@Amruta-Ranade mind taking a first review pass at the changes in #3766 and putting your comments here?

(Happy to do it another way if there is a better/easier way to do this - just trying to keep everyone on the original PR from getting tons of emails / notifications)

**Comment by Amruta-Ranade** [comment]
Excellent idea! Will take a look asap.

**Comment by jseldess** [comment]
@rmloveland, I think it's fine to consider this work done. We can open separate issue when there's internal and external feedback.
